src/main/java/appeng/container/implementations/ContainerStorageBus.java

```diff
@@ -135,7 +139,44 @@ public void detectAndSendChanges() {
                     (StorageFilter) this.getUpgradeable().getConfigManager().getSetting(Settings.STORAGE_FILTER));
             this.setStickyMode((YesNo) this.getUpgradeable().getConfigManager().getSetting(Settings.STICKY_MODE));
         }
+        final int upgrades = this.getUpgradeable().getInstalledUpgrades(Upgrades.CAPACITY);
 
+        if (upgrades > 0) { // sync filter slots
+            boolean needSync = this.storageBus.needSyncGUI;
+            this.storageBus.needSyncGUI = false;
+            if (iRow >= upgrades) {
+                iRow = 0;
+            } else if (needSync) {
+                iRow = upgrades; // to be sure that the last line is sent correctly (do during the next update)
+            } else iRow = iRow + 1;
+
+            IInventory inv = this.getUpgradeable().getInventoryByName("config");
+            int from = needSync ? 18 : 9 + (9 * upgrades); // start at first filter slot or at specific row
+            int to = needSync ? inv.getSizeInventory() : 18 + (9 * upgrades); // end at last filter slot or at end of
+                                                                              // specific row
+            for (; from < to; from++) {
+                if (upgrades <= (from / 9 - 2)) {
+                    break;
+                }
+                ItemStack stack = inv.getStackInSlot(from);
+                if (stack == null) {
+                    continue;
+                }
+                for (ICrafting crafter : this.crafters) {
+                    boolean isChangingQuantityOnly = false;
+                    if (crafter instanceof EntityPlayerMP && ((EntityPlayerMP) crafter).isChangingQuantityOnly) {
+                        isChangingQuantityOnly = true;
+                        ((EntityPlayerMP) crafter).isChangingQuantityOnly = false; // necessary to ensure that the
+                                                                                   // package is sent correctly
+                    }
+                    crafter.sendSlotContents(this, from, stack);
+                    if (isChangingQuantityOnly) {
+                        ((EntityPlayerMP) crafter).isChangingQuantityOnly = true;
```

Comment:
This cast may not be safe since it is outside of the `instanceof EntityPlayerMP` check